File: my-folder/0001-two-sum/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] Result=new int[2];
        HashMap<Integer,Integer> indexes=new HashMap<Integer,Integer>();
        for(int i=0;i<nums.length;i++){
            if(indexes.containsKey(target-nums[i])){
                Result[1]=i;
                Result[0]=indexes.get(target-nums[i]);
                return Result;
            }
        indexes.put(nums[i],i);
        }
    return Result;
    }
}

File: my-folder/0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        indices = {}

        # Two pass approach
        # for idx, num in enumerate(nums):
        #     indices[num] = idx

        # for idx, num in enumerate(nums):
        #     diff = target - num
        #     if diff in indices and indices[diff] != idx:
        #         return[idx, indices[diff]]

        # One pass approach
        for idx, num in enumerate(nums):
            diff = target - num
            if diff in indices:
                return[indices[diff], idx]

            indices[num] = idx
        
